fineract-provider/src/main/java/org/apache/fineract/portfolio/loanproduct/service/LoanProductWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -357,9 +357,8 @@ private void handleDataIntegrityIssues(final JsonCommand command, final Throwabl
             throw new PlatformDataIntegrityException("error.msg.product.loan.duplicate.short.name",
                     "Loan product with short name `" + shortName + "` already exists", "shortName", shortName);
         } else if (realCause.getMessage().contains("Duplicate entry")) {
-            final Object[] args = null;
             throw new PlatformDataIntegrityException("error.msg.product.loan.duplicate.charge",
-                    "Loan product may only have one charge of each type.`", "charges", args);
+                    "Loan product may only have one charge of each type.`", "charges");
```

Comment:
I'm not 100% sure and haven't looked much at this, but could it make sense to:

```suggestion
                    "Loan product may only have one charge of each type.`", "charges", realCause);
```

Reply:
Yes! makes sense :D

Reply:
Also adding the same in rest of the throws in the same function.

Reply:
great, thanks; I've seen it - resolving